### ISSUE TITLE
Extending Native HTML element in Polymer fails in Dartium Dev 1.13

### ISSUE THREAD

**Issue by terrylucas** [dart-lang/sdk#24754]
This simple example will fail:

```
<html>
  <head>
    <script>
      document.registerElement('custom-input', {prototype: Object.create(HTMLInputElement.prototype), extends: 'input'});
    </script>
  </head>
  <body>
    <input is="custom-input" id="myCustomInput" />
    <script type="application/dart">
      import 'dart:html';
      
      main() {
        print(document.getElementById('myCustomInput'));
      }
    </script>
  </body>
</html>
```

**Comment by terrylucas** [comment]
<html>
  <head>
    <script>
      document.registerElement('custom-input', {prototype: Object.create(HTMLInputElement.prototype), extends: 'input'});
    </script>
  </head>
  <body>
    <input is="custom-input" id="myCustomInput" />
    <script type="application/dart">
      import 'dart:html';
      
      main() {
        print(document.getElementById('myCustomInput'));
      }
    </script>
  </body>
</html>

**Comment by zoechi** [comment]
Fail how? Some error message?

**Comment by terrylucas** [comment]
The problem is how wrap_jso constructs a custom element before the Dart class is known.
- If a custom element is extending HTMLElement, before registerElement is called, wrap_jso properly creates a Dart HtmlElement waiting to upgrade this instance to the user's Dart class later (post registerElement call).

Problem #1.  If the custom element is extending a Native element e.g., HTMLInputElement  as in this example.  Then the Dart native element class is not created (InputElement).  Solution is to look walk the __proto__ chain looking for the JS HTML element when found then create the Dart class associated with this JS element - waiting for normal upgrade to user's Dart class to happen.

Problem #2.  Also noticed that auto-binding handling for HTMLTemplateElement works in current Polymer but will fail in Polymer 1.0 (name is dom-binding) again similar problem as above.  Walking the __proto__ chain finding HTMLTemplateElement and creating the Dart class TemplateElement will solve this future problem as well.

Without the fix a null can be returned from wrap_jso.

**Comment by mit-mit** [comment]
@terrylucas, got an update on this one? This is our last scheduled week of 1.13.

**Comment by terrylucas** [comment]
The fix is in 1.13 dev 7.6

commit e84f9941e1cf210646aef72e01eefaa993a7b798
Author: Terry Lucas <terry@google.com>
Date:   Thu Oct 29 17:05:59 2015 -0700
    Fixed Native custom element
    BUG=24754
    R=alanknight@google.com
    Review URL: https://codereview.chromium.org/1422323006 .

**Comment by mit-mit** [comment]
great, thanks!
